COMPOUND_COMPONENTS.md

```diff
@@ -0,0 +1,336 @@
+
+Many other components like `Select`, `Breadcrumbs`, or dropdown menus can also use the `ListModel` and/or the `MenuModel`.
+
+The Typescript interface of a model looks like this:
+```ts
+interface Model<
+  S extends Record<string, any>,
+  E extends Record<string, (...args: any[]) => void
+> {
+  state: S
+  events: E
+}
+```
+    
+The Typescript interface of Actions and Guards looks like this:
+```ts
+type Guard<EventData, State> = ({ data: EventData, state: State}) => boolean
+type Action<EventData, State> = ({ data: EventData, state: State}) => void
+```
+
+
+## Behavior Hooks
+
+### What is a Behavior Hook?
+
+A behavior hook usually applies to a sub-component and describes the events and attributes that are applied to a sub-component's element. A behavior hook takes in the model and developer-defined HTML attributes and return a merged object of events and HTML attributes.
+
+### Why Behavior Hooks?
+
+A behavior hook allows us to more easily reuse functionality between components with similar sub-components. They also provide another layer of composition to compound components. They are usually the accessibility layer of a sub-component and not the HTML element directly.
+
+For example, the `MenuModel` contains model's internal state and events, but doesn't handle external DOM events directly. The behavior hook is the glue between the model and DOM elements. A `useMenu` behavior hook might look like this:
+
+```ts
+const useMenu const useMenu = (
+  model,
+  elemProps = {}
+) => {
+
+  React.useEffect(() => {
+    const item = state.items.find(({ id }) => state.currentId === id)
+    item.ref.current?.focus()
+  }, [state.currentId, state.items])
+
+  return {
+    onKeyDown(event) {
+      // if onKeyDown was provided, call it first
+      elemProps.onKeyDown?.(event)
+
+      if (event.key === 'ArrowLeft') {
+         events.previous()
+      }
+      if (event.key === 'ArrowRight') {
+         events.next()
+      }
+    },
+    ...elemProps,
+  }
+}
+```
+
+## Putting it all together
+
+In the `Tabs` component example, the Menu doesn't actually exist as component, but a model and a behavior hook. The `Tab.List` sub-component uses the `TabModel` and the `useMenu` behavior hook to produce the desired sub-component. It looks something like this:
+
+```tsx
+const TabList = ({children, ...elemProps}) => {
+  const model = useTabsModelContext()
+  const menu = useMenu(
+    model,
+    elemProps,
+  )
+
+  return (
+    <div
+      role="tablist"
+      {...menu}
+    >
+      {children}
+    </div>
+  )
+};
+```
+
+There are 2 ways to configure a model.
+
+The easiest way is to pass model configuration directly to the container component. In the `Tabs` example, it might look like this:
```

Comment:
This example could use a bit of expanding. It's not immediately clear what `onActivateTab` has to do with the model configuration. I think it's just saying something like "props for each model field are generally available for simple customization on components without having to pass in the entire model", etc.

Reply:
I agree. There is a line above that says:
>  A model has configuration for functionality including configuration of events.

But it isn't obvious this is the connection. It should be reiterated.

Reply:
It uses the example above:

https://github.com/Workday/canvas-kit/pull/937/files#diff-a467e1a7dbcaa0ea8af669d262eaab2b94cb1302e12f9d81aaa4117a08872536R185-R214

Reply:
What about the following?

```suggestion
The easiest way is to pass model configuration directly to the container component. Continuing from the `useTabsModel` example above which has a `onActivateTab` action available on the model configuration, it might look like this:
```